packages/bot-skeleton/src/services/tradeEngine/Interface/index.js

```diff
@@ -55,6 +56,10 @@ export default class Interface extends ToolsInterface(TicksInterface(class {}))
             getSellPrice: () => this.getSellPrice(),
             isResult: result => getDetail(10) === result,
             isTradeAgain: result => globalObserver.emit('bot.trade_again', result),
+            emitError: message => {
+                const error = new Error(localize(message));
+                globalObserver.emit('Error', error);
+            },
```

Comment:
we have an  event listener for error messages please take a look at this file `packages/bot-skeleton/src/services/tradeEngine/utils/interpreter.js`

Reply:
Do we have access to it from blockly javascript? Actually I had to add this method to the interface to give access ... please correct me if I am mistaken. Thanks ..